src/routes/utils/helpers.js

```diff
@@ -43,3 +46,13 @@ export const verifyToken = async (req, res, next) => {
     return errorResponse(res, 400, error.message);
   }
 };
+
+export const checkUser = async (req, res, email) => {
```

Comment:
Nice work here. However, i feel this function can be more generic. Instead of taking `req` and `res` as arguments, you can just take email, then return a user if found and throw an error or return null if a user is not found. This way, the function can be outside of a controller or route handler.

Reply:
okay, I will look into this